packages/itwin/tree-widget/src/components/trees/common/UseFeatureReporting.ts

```diff
@@ -48,3 +48,18 @@ export function useFeatureReporting(props: UseFeatureReportingProps): UseFeature
 
   return { reportUsage };
 }
+
+/** @internal */
+export function useReportingAction<T extends (...args: any[]) => void>(
+  action: T,
+  reportUsage?: (props: { featureId?: UsageTrackedFeatures; reportInteraction: true }) => void,
+): T {
+  const reportingAction = useCallback(
+    (...args: any[]) => {
+      reportUsage?.({ reportInteraction: true });
+      action(...args);
+    },
+    [action, reportUsage],
+  );
+  return reportingAction as T;
```

Comment:
Does this work?
```suggestion
  return useCallback<T>(
    (...args: any[]) => {
      reportUsage?.({ reportInteraction: true });
      action(...args);
    },
    [action, reportUsage],
  );
```

Reply:
It does not because `T` can be a subtype of the constraint so the cast is used to force the result type.

Reply:
Looks like this one works. Just need to remove explicit return type.
```suggestion
  return useCallback<(...args: Parameters<T>) => void>(
    (...args) => {
      reportUsage?.({ reportInteraction: true });
      action(...args);
    },
    [action, reportUsage],
  );
```

Reply:
Yes, this one seems to work, changed.